sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/statefulOperators.scala

```diff
@@ -353,6 +353,7 @@ case class StateStoreSaveExec(
                   finished = true
                   null
                 } else {
+                  numOutputRows += 1
```

Comment:
A regression introduced in https://github.com/apache/spark/pull/18107 ?

Reply:
Yes, looks like so. The API seemed to be revised completely in #18107 and I don't have background though.